src/vecdb/vecdb.rs

```diff
@@ -2,52 +2,187 @@ use std::path::PathBuf;
 use std::sync::Arc;
 
 use async_trait::async_trait;
-use tokio::sync::Mutex;
+use tracing::info;
+use tokio::sync::Mutex as AMutex;
 use tokio::task::JoinHandle;
+use crate::global_context::{CommandLine, GlobalContext};
+use tokio::sync::RwLock as ARwLock;
+use tracing::error;
+use crate::background_tasks::BackgroundTasksHolder;
 
 use crate::vecdb::handler::{VecDBHandler, VecDBHandlerRef};
-use crate::vecdb::req_client::get_embedding;
+use crate::fetch_embedding::get_embedding;
 use crate::vecdb::vectorizer_service::FileVectorizerService;
 use crate::vecdb::structs::{SearchResult, VecdbSearch, VecDbStatus};
 
+
 #[derive(Debug)]
 pub struct VecDb {
     vecdb_handler: VecDBHandlerRef,
-    retriever_service: Arc<Mutex<FileVectorizerService>>,
-    embedding_model_name: String,
-    cmdline: crate::global_context::CommandLine,
+    retriever_service: Arc<AMutex<FileVectorizerService>>,
+    cmdline: CommandLine,
+
+    model_name: String,
+    api_key: String,
+    cloud_name: String,
+    endpoint_template: String,
```

Comment:
api_key is the only thing that doesn't change.

cloud_name should not exist at all.

model_name, endpoint_template -- changes with caps. A pointer to caps is better.